dts/xtensa/intel/intel_adsp_ace15_mtpm.dtsi

```diff
@@ -277,6 +277,24 @@
 			status = "okay";
 		};
 
+		dmic0: dai-dmic@10000 {
+			compatible = "intel,dai-dmic";
+			reg = <0x10000 0x8000>;
+			fifo = <0x0008>;
```

Comment:
Still nitpicky: you dropped the second item, but now it's an array of one element.  You can have integer properties ("type: int"), which seems like what you really want.

Also I still think this would be clearer if it stored an index (0 or 1) instead of a register offset.

(Not a -1, just a note)

Reply:
np, good to get feedback. I just noticed that zephyr is silently merging the 2 nodes with same register address.. ??? So I don't get 2 devices, only 1. So I probably need to come up with some other scheme here for the nodes.

Reply:
> So I probably need to come up with some other scheme here for the nodes.

Assuming that they really are two different peripherals that just happen to have the same register map addresses, you can just give them different names. Examples:

https://github.com/zephyrproject-rtos/zephyr/blob/6f9e8f459d261f36dc4cf7e50376f4a4931e39ea/dts/arm/nordic/nrf52840.dtsi#L87
https://github.com/zephyrproject-rtos/zephyr/blob/6f9e8f459d261f36dc4cf7e50376f4a4931e39ea/dts/arm/nordic/nrf52840.dtsi#L104

This will cause a warning which you can disable with https://github.com/zephyrproject-rtos/zephyr/blob/6f9e8f459d261f36dc4cf7e50376f4a4931e39ea/boards/arm/nrf52840dk_nrf52840/pre_dt_board.cmake#L7.

Reply:
... although you did just find a bug in dtlib. I assumed `dtc` would have the same behavior, but it actually treats this as an error by default:

```
$ cat foo.dts
/dts-v1/;

/ {
	blub: foo@1234 {
		reg = <1234 4>;
		bar = <1>;
	};

	blab: foo@1234 {
		reg = <1234 4>;
		baz = <2>;
	};
};

$ dtc -Odts foo.dts
foo.dts:8.17-10.4: ERROR (duplicate_node_names): /foo@1234: Duplicate node name
ERROR: Input tree has errors, aborting (use -f to force output)
```

We are compatible with dtc by contract, so this should have been an error. I'll file a bug, thank you for noticing this.

Reply:
@mbolivar-nordic I tried to do this as you suggested... hopefully kind of what you were asking